src/Security/Authentication/JwtBearer/src/JwtBearerHandler.cs

```diff
@@ -208,6 +219,47 @@ protected override async Task<AuthenticateResult> HandleAuthenticateAsync()
         }
     }
 
+    private void RecordTokenValidationError(Exception exception, List<Exception> exceptions)
+    {
+        if (exception != null)
+        {
+            Logger.TokenValidationFailed(exception);
+            exceptions.Add(exception);
+        }
+
+        // Refresh the configuration for exceptions that may be caused by key rollovers. The user can also request a refresh in the event.
+        // Refreshing on SecurityTokenSignatureKeyNotFound may be redundant if Last-Known-Good is enabled, it won't do much harm, most likely will be a nop.
+        if (Options.RefreshOnIssuerKeyNotFound && Options.ConfigurationManager != null
+            && exception is SecurityTokenSignatureKeyNotFoundException)
+        {
+            Options.ConfigurationManager.RequestRefresh();
+        }
+    }
+
+    private async Task<TokenValidationParameters> SetupTokenValidationParametersAsync()
+    {
+        // Clone to avoid cross request race conditions for updated configurations.
+        var tokenValidationParameters = Options.TokenValidationParameters.Clone();
+
+        if (Options.ConfigurationManager is BaseConfigurationManager baseConfigurationManager)
+        {
+            tokenValidationParameters.ConfigurationManager = baseConfigurationManager;
+        }
+        else
+        {
+            if (Options.ConfigurationManager != null)
+            {
+                // GetConfigurationAsync has a time interval that must pass before new http request will be issued.
+                var configuration = await Options.ConfigurationManager.GetConfigurationAsync(Context.RequestAborted);
```

Comment:
Previously we would cache the `_configuration` on the handler. However, we don't cache it anymore, and instead will always call `GetConfigurationAsync`. Is this OK? Will it cause any sort of performance issue?

Reply:
GetConfigurationAsync should do its own caching, and handlers are re-created per request anyways. It looks like this is the only place config is read?

Reply:
Correct, ConfigurationManager does its own caching